project/Mibudin/README.md

```diff
@@ -0,0 +1,251 @@
+# WFCpp: <br>a Simple C++ Library for Wave Function Collapse
+
+> [**WFCpp**][wfcpp-gh-repo] is a simple library for [Wave Function
+> Collapse][wfc] implemented with modern C++ and also provides interfaces to
+> Python with [pybind11][pybind11].
+
+
+## Basic Information
+
+- **Project Name**:
+    WFCpp
+- **GitHub Repository**:
+    [https://github.com/Mibudin/wfcpp][wfcpp-gh-repo]
+- **Languages**:
+    C++, Python (with [pybind11][pybind11])
+
+Procedural modeling of 2D textures or 3D models is a famous and popular topic
+in the field of Computer Graphics. Among various techniques, [**Wave Function
+Collapse**][wfc] is one of the most well-known algorithm, which majorly used
+and popularized a previous academical work, [Model Synthesis][model-synthesis].
+Please refer to Wave Function Collapse's [GitHub repository][model-synthesis]
+where provides awesome description of this method. This algorithm has been used
+by many game developers. By the way, it is originally a kind of 2D texture
+synthesis techniques, but it can be also extended to 3D models[^3D-wfc-impl].
+
+There is a long tradition of requiring "**local similarity**"[^local-similar]
+when doing texture synthesis. Many algorithm of procedural modeling uses
+"**grammar**"[^grammar] to produce models that are local similar to the input
+one. Wave Function Collapse uses such grammars from the pixel/grid/tile-based
+example to produce various results.
+
+
+## Problem to Solve
+
+The primary topic of Wave Function Collapse is to use the **grammar** with the
+**pixel/grid/tile-based example** as the input and then iteratively generate
+result graphics by **propagating** "waves" to analyze the possible results of
+the remaining undetermined. The author called this concept as "minimal entropy
+heuristic" way. This project needs to implement these essential functionalities
+to make usage of the grammars and finally achieve the final goal.
+
+Overall, there exist enormous and various implementing ways of Wave Function
+Collapse which differ in detail of the implementation or use-customized
+specifications in order that they can focus on different fields, such as
+parallelization, result preferences, etc. Therefore, it is needed to determine
+or select an appropriate implementation version among them.
+
+Besides, this project has to provide friendly and widely flexible interfaces
+between the C++ library and Python codes, to make this project easy to be used
+in many kinds of applications.
+
+
+## Prospective Users
+
+The people who are interested or needed to use Wave Function Collapse as a
+procedural modeling method to produce large-scale textures or models that are
+local similar to given pixel/grid/tile-based examples. They may be artist,
+designers, game developers, etc.
+
+
+## API Description
+
+> [!WARNING]
+> The following API description only focuses on major exposed C++ API, which is
+> brief and may not be the final version.
```

Comment:
Good.  You do not need to stick with the initial proposal when developing the real code.  It may be a good learning to see how differently you end up with.